rust/arrow/src/datatypes.rs

```diff
@@ -618,7 +628,7 @@ macro_rules! make_numeric_type {
             fn mask_from_u64(mask: u64) -> Self::SimdMask {
                 match Self::lanes() {
                     8 => {
-                        let vecidx = i64x8::new(128, 64, 32, 16, 8, 4, 2, 1);
+                        let vecidx = i64x8::new(1, 2, 4, 8, 16, 32, 64, 128);
```

Comment:
Tests for min/max exposed that these constants were actually in the wrong order. This didn't turn up in the sum tests because there all invalid elements were already set to 0. I think when I initially implemented this, I copied the initialization from code that used intel avx intrinsics which take the parameters in the opposite order :facepalm:

Reply:
And the tests for mask generation were apparently also incorrect.

Reply:
Maybe a line comment before this assignment could help others in understanding these numbers?